.pre-commit-config.yaml

```
@@ -21,3 +23,17 @@ repos:
       language: lua
       files: 'Poké-data.lua|AltForms-data.lua|UselessForms-data.lua|BothForms-data.lua'
       pass_filenames: false
+
+    - id: changed-snapshot-tests
+      name: Run snapshot tests on changed files
+      entry: bash test/pre-commit-snapshots.sh
+      language: lua
+      files: '.*\.lua$'
```

```suggestion
      files: ^test/
      types: [lua]
```
On the pre-commit side of things I think this is what you wanted to do. Using `types` instead of `files` regex [is preferred](https://pre-commit.com/#filtering-files-with-types). The regex on `files` now limits the passed filenames to modified test files.

On the conceptual side, this pre-commit re-runs tests whose _source_ changed, while we would like to re-run those whose _output_ may have changed due to other files changing. I understand this is hard to detect statically tho (would need some kind of lua dependency resolution), and that we catch those at `pre-push`, so this is probably as good as we get.

---

I actually meant to run tests whose *source* changed. It can serve as a form of smoke testing for trivial bugs, and it's actually pretty decent for changes to modules at the end of the dependency chain (e.g. `DebRes`).  
But yeah, we should definitely check the tests on commit if the tests have hanged too.